### PR TITLE
Fix paragraph-awareness of the composer formatting features

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/11882

### CODE REVIEW

File: test/editor/operations-test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {getLineAndNodePosition} from "../../src/editor/caret";
+import EditorModel from "../../src/editor/model";
+import {createPartCreator, createRenderer} from "./mock";
+import {toggleInlineFormat} from "../../src/editor/operations";
+
+describe('editor/operations: formatting operations', () => {
```
Comment (bwindels) [inline test/editor/operations-test.js]: awesome, thank you for writing more tests!

**Comment by t3chguy** [comment]
Oh whoops, it looks like I forgot to push the latest changes which I think address most of your concerns

File: src/editor/operations.js [hunk 2]
```diff
@@ -104,23 +104,63 @@ export function toggleInlineFormat(range, prefix, suffix = prefix) {
...
+    // compute paragraph [start, end] indexes
+    const paragraphIndexes = [];
+    let startIndex = 0;
+    // let seenNewlines = 0;
+    for (let i = 2; i < parts.length; i++) {
+        // paragraph breaks can be denoted in a multitude of ways,
+        // - 2 newline parts in sequence
+        // - newline part, plain(<empty or just spaces>), newline part
+
+        const isBlank = part => !part.text || !/\S/.test(part.text);
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/editor/operations.js]: This already happened, you somehow reviewed a stale diff: https://github.com/matrix-org/matrix-react-sdk/pull/3891/files#diff-a901c955a82fb6a94975067275b941e6R104

File: src/editor/operations.js [hunk 3]
```diff
@@ -104,23 +104,63 @@ export function toggleInlineFormat(range, prefix, suffix = prefix) {
...
+        const isNL = part => part.type === "newline";
+
+        // bump startIndex onto the first non-blank after the paragraph ending
+        if (isBlank(parts[i - 2]) && isNL(parts[i - 1]) && !isNL(parts[i]) && !isBlank(parts[i])) {
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply src/editor/operations.js]: It overwrites startIndex after the paragraph indexes are stored and advances it to the next part after the end of this paragraph as an assumption of where the next paragraph begins

**Comment by bwindels** [comment]
@t3chguy Shall I wait for you to push mentioned changes? Currently, I can only see the comment before the loop being added.

File: src/editor/operations.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bwindels) [reply src/editor/operations.js]: Oh I see, lgtm.

**Pull request merged**